libraries/maven/maven.groovy

```diff
@@ -0,0 +1,45 @@
+package libraries.maven
+
+void call(){
+
+}
+
+void run(Map params = [:], String phase) {
+    if (!config.mavenId) {
+        error "Must supply the installed Maven version's ID"
+    }
+    this.run(phase, params.get('goals', []) as ArrayList<String>, params.get('properties', [:]) as Map<String, String>, params.get('profiles', []) as ArrayList<String>)
+}
+
+// Run maven command with installed maven version
+void run(String phase, ArrayList<String> goals, Map<String, String> properties, ArrayList<String> profiles) {
+    stage("Maven") {
+        withMaven(maven: config.mavenId) {
+            String command = "mvn clean "
```

Comment:
S: Should it be invoking clean here?

Reply:
I assumed it was safer since some plugins may require it. I don't think it can cause problems even if it's not the most efficient route unless I'm mistaken.

Reply:
I think we should have them specify it. Sometimes you don't want it cause it will break things - like release.